docs/DeveloperGuide.md

```diff
@@ -248,10 +237,157 @@ Here is an example of a `Person` in JSON form:
+<br>
+In this category of commands, the class`PersonAdapter` is doing most of the legwork here.
+`PersonAdpater` serves as a wrapper for the Model class `Person`.
+The key differences are that `Person` is immutable and does not support edits, while the `PersonAdapter` effectively supports edits by wrapping a single `Person` object and replacing it with an edited copy as and when necessary.
+Such an implementation supports the user viewing and controlling a single client like with the `ViewCommand`.
+</div>
+
+#### 4.2.3 Commands that involve changing of tabs
+
+<div markdown="span" class="alert alert-info">:information_source: **Key Properties:** 
+<br>
+The key differences are:<br>
+1. The commands do not interact with model.<br>
+2. The commands have to handle the changing of ParserStrategy, from the current one<br>
+to `ABCParser` of the new tab.
+3. The commands need to update the `UiState` of `Ui` to keep track of which tab the user is currently on.
+<br>
+<br>
+Commands that fall into this category are:<br>
+1. SwitchTab<br>
+2. View<br>
+</div>
+
+Switch tab command is a command that explicitly involves the changing of tabs, which fall under the second [category](#42-implementation-and-commands)
+We will be using the `SwitchTab` command as the example to illustrate and explain all commands under this category.
+
+<img alt="SwitchTabExecution" src="images/SwitchTabExecutionSequenceDiagram.png" width="600"/>
+
+Explanation of diagram above:
+* The `UI` takes in the command inputted from the user and passes it to the `Logic` component that is responsible for parsing the input.
+* The `Logic` component parses the command and returns the `SwitchTab` command.
+* The `Logic` component executes the `SwitchTab` command and returns the `SwitchTabCommandResult`
+* The `Logic` component then accepts the LogicConsumer produced from the `SwitchTabCommandResult`.
+In this case, for the `SwitchTab` command, a new `ParserStrategy` is set here. 
+* The `UI` component then accepts the UiConsumer produced from the command result. UiState is set here. 
+
+
+<div markdown="span" class="alert alert-warning">**Explanation of ParserContext:**
+<br>
+1. When a `XYZCommand` class (e.g. `HomeCommand`, `ViewCommand`,...) is executed, it returns a `CommandResult` object containing a logic action if the `XYZCommand` requires a change in tab or view. <br>
+2. `LogicManager` accepts this `CommandResult` object and executes the logic action here.`LogicManager` is a facade that is able to set and change the current `ParserStrategy`.<br>
+3. `ParserContext` in `LogicManager` is updated to contain the `ABCParser` of the new view or tab.<br>
+4. `UI` is updated to change its state, which is kept track of by `UiState` by accepting the consumer also in the command result.<br>
+</div>
+
+#### 4.3 NotesTab
+
+The notes tab is different from the commands above. Instead of being command-based, the notes tab allows the user
```

Comment:
a tab being different from the command? might want to rephrase this para

Reply:
Rephrased. The notes tab accepts user input like the commands but is fundamentally different from it.